### ISSUE TITLE
How to export SPP-NET to onnx ?

### ISSUE THREAD

**Issue by yongxin3344520** [pytorch/pytorch#60253]
Here's my code：
----------------------------------------------------------start----------------------------------------------------------------
import torch
import torch.nn as nn
import math
import torch.nn.functional as F

class Spp(nn.Module):

    def __init__(self, level, pooling_type="max_pool"):
        super().__init__()
        self.num = level
        self.pooling_type = pooling_type

    def forward(self, x):
        h, w = x.shape[2:]
        kernel_size = (math.ceil(h / self.num), math.ceil(w / self.num))
        stride = kernel_size
        pooling = (math.ceil((kernel_size[0] * self.num - h) / 2), math.ceil((kernel_size[1] * self.num - w) / 2))
        if self.pooling_type == 'max_pool' or self.pooling_type == "max":
            tensor = F.max_pool2d(x, kernel_size=kernel_size, stride=stride, padding=pooling)
        else:
            tensor = F.avg_pool2d(x, kernel_size=kernel_size, stride=stride, padding=pooling)
        return tensor

class SppNet(nn.Module):
    
    def __init__(self, pooling_type="max_pool", level=1):
        super(SppNet, self).__init__()
        self.spps = []
        for i in range(level):
            self.spps.append(Spp(pooling_type=pooling_type, level=i+1))
        pass

    def forward(self, x):
        n, c = input.shape[0:2]
        out = []
        for spp in self.spps:
            y = spp(x).reshape(n, c, -1)
            out.append(y)
        out = torch.cat(out, dim=2)
        return out

if __name__ == '__main__':
    input = torch.randn(3, 45, 100, 120)
    sppNet = SppNet(level=7)
    y0 = sppNet(input)
    print(y0.shape)

    sppNet.eval()

    torch.onnx.export(sppNet,  # model being run
                      input,  # model input (or a tuple for multiple inputs)
                      'spp-net.onnx',
                      # where to save the model (can be a file or file-like object)
                      export_params=True,  # store the trained parameter weights inside the model file
                      opset_version=11,  # the ONNX version to export the model to
                      do_constant_folding=True,  # whether to execute constant folding for optimization
                      input_names=["input"],  # the model's input names
                      output_names=["output"],  # the model's output names
                      dynamic_axes={
                          "input": {0: "batch_size", 1: "channel", 2:"height", 3:"width"},
                          "output": {0: "batch_size", 1: "channel", 2:"length"}
                      },
                      enable_onnx_checker=True)

------------------------------------------------------end----------------------------------------------------------------

Exported model  “kernel_ size、pooling ” parameter is fixed. 
But I need to set it to a fixed parameter. 
That is to say, the kernel is calculated automatically according to the input size and other parameters, so how to do?
Ask for advice，Thank  you!

cc @garymm @BowenBao @neginraoof

**Comment by BowenBao** [comment]
> That is to say, the kernel is calculated automatically according to the input size and other parameters, so how to do?

@yongxin3344520 Unfortunately, you can not. `kernel_size` and `pads` for ONNX `MaxPool` operator are static by spec https://github.com/onnx/onnx/blob/master/docs/Operators.md#MaxPool. There is no way to export an ONNX model that computes with dynamic kernel size with standard Pooling operators.

**Comment by shubhambhokare1** [comment]
Hi @yongxin3344520 ,

We’ve gone ahead and closed this issue because it is stale.
If you still believe this issue is relevant, please feel free to reopen the issue and we will triage it as necessary. Please specify in a comment any updated information you may have so that we can address it effectively. We encourage you to try the latest pytorch-preview (nightly) version to see if it has resolved the issue.

Thanks,
ONNX Converter team
